src/ucp/wireup/wireup_cm.c

```diff
@@ -271,8 +272,11 @@ static unsigned ucp_cm_client_connect_progress(void *arg)
 
     ucs_assert(addr.address_count <= UCP_MAX_RESOURCES);
     ucs_assert(wireup_ep->ep_init_flags & UCP_EP_INIT_CM_WIREUP_CLIENT);
-    status = ucp_wireup_init_lanes(ucp_ep, wireup_ep->ep_init_flags, &addr,
-                                   addr_indices);
+    tl_bitmap = ucp_ep_get_tl_bitmap(ucp_ep);
```

Comment:
how do we have tl bitmap before lanes are initialized?

Reply:
from key, it's initialized

Reply:
but at this point ep should have only CM lane? so key is not real transport yet?

Reply:
should be "real" for homogenous set up, but must be correct from device perspective.
seems like need to retrieve device from previous configuration and recalculate bitmap for all TLs, WDYT?

Reply:
done